src/Main.java

```diff
@@ -1585,22 +1620,45 @@ else if (location.equals("W3G")) {
                     && canSwim(inventory);
         }
         else if (location.equals("W3B")) {
+			/*
+			* Added MERCILESS logic to this chest.
```

Comment:
This is correct. Once you walk off the top pipe, if you've done everything correct up to that point, you will warp to the bottom of the pool with the block lifted up as if you had the air pump. However, it's treated as if you're still on the surface, thus you get kind of mixed properties on it. Since you're right on the door tiles after the warp, you can press Up again and you will enter the Blue Chest room.